### PR TITLE
Remove more unused attributes in config classes

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Remove more unused attributes in config classes.

There are more changes needed than I expected previously. I have to adopt the new test (now only in my branch) in a progressive way and make the changes to pass that test in the same time.

I tried to change more places in single PR to avoid too many PRs related to this topic. But there will still be a few PRs in the future 🙏 .

### CODE REVIEW

File: src/transformers/models/altclip/configuration_altclip.py [hunk 1]
```diff
@@ -77,12 +77,8 @@ class AltCLIPTextConfig(PretrainedConfig):
         use_cache (`bool`, *optional*, defaults to `True`):
             Whether or not the model should return the last key/values attentions (not used by all models). Only
             relevant if `config.is_decoder=True`.
-        classifier_dropout (`float`, *optional*):
```
Comment (ydshieh) [inline src/transformers/models/altclip/configuration_altclip.py:80]: `hidden_dropout_prob` and `attention_probs_dropout_prob` are used, but not this one.

File: src/transformers/models/altclip/configuration_altclip.py [hunk 2]
```diff
@@ -175,8 +167,6 @@ class AltCLIPVisionConfig(PretrainedConfig):
             The non-linear activation function (function or string) in the encoder and pooler. If string, `"gelu"`,
             `"relu"`, `"selu"` and `"gelu_new"` ``"quick_gelu"` are supported. layer_norm_eps (`float`, *optional*,
             defaults to 1e-5): The epsilon used by the layer normalization layers.
-        dropout (`float`, *optional*, defaults to 0.0):
```
Comment (ydshieh) [inline src/transformers/models/altclip/configuration_altclip.py:178]: `hidden_dropout_prob` and `attention_probs_dropout_prob` are used, but not this one.

File: src/transformers/models/biogpt/configuration_biogpt.py [hunk 3]
```diff
@@ -111,7 +111,6 @@ def __init__(
         layer_norm_eps=1e-12,
         scale_embedding=True,
         use_cache=True,
-        is_encoder_decoder=False,
```
Comment (ydshieh) [inline src/transformers/models/biogpt/configuration_biogpt.py:114]: `is_encoder_decoder` may not appear in modeling file, but it's important in generation files.
The default value in `PretrainedConfig` is `False`. Let's not add it here to make thing a bit easier/cleaner.

File: src/transformers/models/bridgetower/configuration_bridgetower.py [hunk 4]
```diff
@@ -311,12 +309,10 @@ class BridgeTowerConfig(PretrainedConfig):
     def __init__(
         self,
         share_cross_modal_transformer_layers=True,
-        drop_rate=0.1,
```
Comment (ydshieh) [inline src/transformers/models/bridgetower/configuration_bridgetower.py:314]: not used. cc @NielsRogge

File: src/transformers/models/ctrl/configuration_ctrl.py [hunk 5]
```diff
@@ -96,11 +96,6 @@ def __init__(
         embd_pdrop=0.1,
         layer_norm_epsilon=1e-6,
         initializer_range=0.02,
-        summary_type="cls_index",
```
Comment (ydshieh) [inline src/transformers/models/ctrl/configuration_ctrl.py:99]: These may not appear in modeling file, but they are necessary if the model use
```python
self.sequence_summary = SequenceSummary(config)
```
For `CTRL` model (and some other models below), they don't use `SequenceSummary`. So remove those attributes here.

cc @NielsRogge

File: src/transformers/models/whisper/configuration_whisper.py [hunk 6]
```diff
@@ -129,8 +129,6 @@ class WhisperConfig(PretrainedConfig):
             Begin of stream token id.
         eos_token_id (`int`, *optional*, defaults to 50257):
             End of stream token id.
-        tie_word_embeddings (`bool`, *optional*, defaults to `True`):
```
Comment (ydshieh) [inline src/transformers/models/whisper/configuration_whisper.py:132]: The default value in `PretrainedConfig` is `True`. We don't need to specify here. Removing here just for housekeeping. cc @ArthurZucker

**Review by ydshieh — COMMENTED** [review]
I leave a few comments to explain the reason of removing those attributes.

**Review by sgugger — APPROVED** [review]
Thanks for the cleanup!

**Review by sgugger — APPROVED** [review]
Thanks for the clean-up!

**Review by younesbelkada — APPROVED** [review]
LGTM on the BLIP side, thanks for working on this 💯

**Pull request merged**